pkg/merger/merger_test.go

```diff
@@ -408,6 +408,71 @@ func Test_MergeAndUpdate(t *testing.T) {
 		})
 	}
 }
+func Test_RecordLastModified(t *testing.T) {
+	cases := []struct {
+		name   string
+		attrs  *storage.ReaderObjectAttrs
+		mets   *Metrics
+		source string
+	}{
+		{
+			name:   "nil attrs; succeeds",
+			attrs:  nil,
+			mets:   &Metrics{},
+			source: "fake-source",
+		},
+		{
+			name:   "nil mets; succeeds",
+			attrs:  &storage.ReaderObjectAttrs{},
+			mets:   nil,
+			source: "fake-source",
+		},
+		{
+			name:   "nil; succeeds",
+			attrs:  nil,
+			mets:   nil,
+			source: "",
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			recordLastModified(tc.attrs, tc.mets, tc.source)
+		})
+	}
+}
+
+func Test_RecordLastModified_RecordMetrics(t *testing.T) {
+	var fakeMetric FakeInt64
+
+	attrs := &storage.ReaderObjectAttrs{
+		LastModified: time.Now(),
+	}
+	mets := &Metrics{
+		LastModified: &fakeMetric,
+	}
+	source := ""
+
+	recordLastModified(attrs, mets, source)
+	if fakeMetric.last != 0 {
+		t.Errorf("recordLastModified() fail; got %d, want %d", fakeMetric.last, 0)
+	}
+}
+
+type FakeInt64 struct {
+	read bool // Fake implementation not concurrent-safe
```

Comment:
Seems like you ought to make it concurrent safe (at least if this will be used outside this package).

Reply:
I think it would be good to rename this to a "fakeInt64" to make it private.

Keeping things package-scoped until you need them outside the package is generally good, since private -> public is easier than public -> private.